verify-docker-versions

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set -o pipefail
+
+# Assuming docker minor and patch version backward-compatibility,
+# but not major version compatibility (require 1.X.*).
+
+VERSION_MINOR_MIN_DOCKER_ENGINE="11"
+VERSION_MINOR_MIN_DOCKER_COMPOSE="6"
+
+compare_versions() {
+  VERSION_ACTUAL=$1
+  VERSION_MINIMUM=$2
+
+  if [[ -z $VERSION_ACTUAL ]]; then
+    echo "ERROR: Actual version is empty (hint: is it installed, or version format change?)."
+    return 1
+  fi
+
+  if [ "$VERSION_ACTUAL" -lt "$VERSION_MINIMUM" ]; then
+    echo "ERROR: Actual version installed ($VERSION_ACTUAL) is smaller than minimum requirement ($VERSION_MINIMUM)."
+    return 1
+  fi
+
+  return 0
+}
+
+
+# test Docker Engine version
+
+DOCKER_ENGINE_VERSION_REGEX_MINOR_BACKREF="Docker version 1\.([0-9]+)\.[0-9]+, build [0-9a-f]+"
+
+DOCKER_ENGINE_VERSION_STRING=$(docker --version 2>/dev/null | head -n1)
```

Comment:
I would replace this with `docker info`:

```
mborzecki:mendersoftware/deployments/tests$ docker info -f '{{.ServerVersion}}'
1.13.1
```

`docker info` pings docker daemon. AFAIK it's possible to have a different `docker` version A locally and talk to a remote daemon in version B. Another advantage is that if someone is using `DOCKER_HOST` we'll ping the right daemon.